solc/CommandLineInterface.cpp

```diff
@@ -594,6 +594,29 @@ bool CommandLineInterface::compile()
 
 		m_compiler->enableIRGeneration(m_options.compiler.outputs.ir || m_options.compiler.outputs.irOptimized);
 		m_compiler->enableEwasmGeneration(m_options.compiler.outputs.ewasm);
+		m_compiler->enableEvmBytecodeGeneration(
+			m_options.compiler.outputs.asm_ ||
+			m_options.compiler.outputs.asmJson ||
+			m_options.compiler.outputs.opcodes ||
+			m_options.compiler.outputs.binary ||
+			m_options.compiler.outputs.binaryRuntime ||
+			m_options.compiler.outputs.metadata ||
+			(
+				m_options.compiler.combinedJsonRequests && (
+					m_options.compiler.combinedJsonRequests->metadata ||
+					m_options.compiler.combinedJsonRequests->binary ||
+					m_options.compiler.combinedJsonRequests->binaryRuntime ||
+					m_options.compiler.combinedJsonRequests->opcodes ||
+					m_options.compiler.combinedJsonRequests->asm_ ||
+					m_options.compiler.combinedJsonRequests->generatedSources ||
+					m_options.compiler.combinedJsonRequests->generatedSourcesRuntime ||
+					m_options.compiler.combinedJsonRequests->srcMap ||
+					m_options.compiler.combinedJsonRequests->srcMapRuntime ||
+					m_options.compiler.combinedJsonRequests->funDebug ||
+					m_options.compiler.combinedJsonRequests->funDebugRuntime
+				)
+			)
+		);
```

Comment:
Is that really enough? I think that for example `--gas` requires EVM compilation too. At least in Standard JSON `evm.gasEstimates` does enable the old codegen:

https://github.com/ethereum/solidity/blob/8447b32d52d5fb965e1a0ff696e8c52634dd4b76/libsolidity/interface/StandardCompiler.cpp#L278-L294

Also, in Standard JSON this check does not include `opcodes` and `metadata`. Is that a bug?

I think it might be a good idea to unify these two.

Reply:
``metadata`` is indeed not needed - ``opcodes`` might just not be available with Standard JSON, but I'm not sure.
And yes, for sure we should unify this - that's why I keep saying, that the CLI should be rewritten as thin wrapper around ``StandardCompiler`` ;-).

Reply:
Also the gas estimates indeed need this - I added them to the list.

Reply:
> `opcodes` might just not be available with Standard JSON

I just tried `["evm.bytecode.opcodes"]` and it seems to work fine. Looks like there are some extra components are hidden in `evmObjectComponents()`:
- For `bytecode`: `object`, `opcodes`, `sourceMap`, `functionDebugData`, `generatedSources`, `linkReferences`
- For  `deployedBytecode`: `object`, `opcodes`, `sourceMap`, `functionDebugData`, `generatedSources`, `linkReferences`,  `immutableReferences`

So it does generate EVM bytecode for `opcodes` after all. `linkReferences` and  `immutableReferences` seem to be the only things you are missing but CLI does not support printing these currently as far as I can tell. So at least your list now matches Standard JSON, but, as @chriseth has pointed out, if will be very easy to forget about updating this if we ever add these extra keys in the CLI.

By the way, it  looks like output selection in Standard JSON is not completely validated. Invalid values like like `["evm.opc"]` or `["evm.abcdef"]` is not an error. Is this for compatibility or should we start validating them?